wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.kt

```
@@ -1290,13 +1310,33 @@ class BlockchainServiceImpl : LifecycleService(), BlockchainService {
     }
 
+    private fun scheduleRetryForegroundService(notification: Notification) {
+        pendingForegroundNotification = notification
+        // Schedule a retry after a few seconds to see if the app comes to foregrxound
+        handler.postDelayed({
+            if (pendingForegroundNotification != null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                try {
+                    startForeground(pendingForegroundNotification!!)
+                    pendingForegroundNotification = null
+                    log.info("Successfully started foreground service on retry")
+                } catch (e: ForegroundServiceStartNotAllowedException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)